### PR TITLE
minor fixes

### PR DESCRIPTION
**Pull request by ibrokethecloud**
PR introduces minor fixes for the following:

* logout method calls when using redfish api to query hardware details / events
* trigger reconcile of underlying machine crd when inventory spec changes.

### CODE REVIEW

**Review by bk201 — APPROVED** [review]
LGTM!

**Review by guangbochen — APPROVED** [review]
LGTM, thanks.

**Comment by ibrokethecloud** [comment]
lastest commit introduces a validating webhook to identify duplicate inventory objects.

File: pkg/controllers/setup.go [hunk 1]
```diff
@@ -166,12 +168,18 @@ func (s *Server) Start(ctx context.Context) error {
...
+	eg, egCtx := errgroup.WithContext(ctx)
+	eg.Go(func() error {
+		s.logger.Info("starting manager")
+		return mgr.Start(egCtx)
+	})
+
+	eg.Go(func() error {
+		s.logger.Info("starting webhook")
+		return webhook.SetupWebhookServer(egCtx, mgr, s.LeaderElectionNamespace)
+	})
```
Comment (guangbochen) [inline pkg/controllers/setup.go:180]: Question: any pros and cons of running webhook as a thread vs running as a separated container/pod?

File: pkg/webhook/setup.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
+package webhook
+
+import (
+	"context"
+	"time"
+
+	seederv1alpha1 "github.com/harvester/seeder/pkg/api/v1alpha1"
```
Comment (guangbochen) [inline pkg/webhook/setup.go:7]: ditto

File: pkg/controllers/setup.go (hunk 1 above)
Comment (ibrokethecloud) [reply pkg/controllers/setup.go:180]: simplified packaging. No additional docker repos need to be setup etc.

plus the webhook is only going to serve a minimal change.

File: pkg/webhook/inventory_validator.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
+package webhook
+
+import (
+	"context"
+	"fmt"
+
+	seederv1alpha1 "github.com/harvester/seeder/pkg/api/v1alpha1"
```
Comment (bk201) [inline pkg/webhook/inventory_validator.go:7]: It would be nice to put same-module import to the last.

**Review by guangbochen — APPROVED** [review]
LGTM, the webhook is working correctly, and please squash the commits before merging, thanks.
@WuJun2016 @n313893254 For the UI, can we consider skipping updating the storage config if there is no update? thanks
image

**Comment by WuJun2016** [comment]
> LGTM, the webhook is working correctly, and please squash the commits before merging, thanks. @WuJun2016 @n313893254 For the UI, can we consider skipping updating the storage config if there is no update? thanks image

We plan to enhance this message prompt in the next release.

**Review by starbops — APPROVED** [review]
LGTM, thanks!

**Pull request merged**